### DISCUSSION TITLE
How to make dynamic routes to ignore static file paths from /public/ folder?

### DISCUSSION THREAD

**Discussion by finom** [vercel/next.js#22224] [Help] [answered]
I use "catch-all" route at project root. 
![image](https://user-images.githubusercontent.com/1082083/108077785-98161280-7075-11eb-8402-b067c1bcb374.png)
Unfortunately it also overrides "public" folder logic and catches all requests which supposed to return a static file.
```
/pubic/
  /foo.txt
```

http://localhost:3000/foo.txt returns what is caught by `[[...path]].tsx` file but not the static file. Can you please help me to solve that and make NextJS always return static files even though if I use dynamic routes?

**Comment by mkrtchian** [comment, marked as answer]
Hello, I just tried to start an app with the following command:
```bash
npx create-next-app --example with-typescript
```
Then I renamed `index.tsx` to `[[...path]].tsx`, created a `public/foo.txt`, and http://localhost:3000/foo.txt displays the content of foo.txt.

**Comment by michrome** [comment]
Files in the `public` directory take priority over dynamic routes.

**Comment by finom** [comment]
Sorry guys, that's my mistake. I thought if I use /src/ folder then /public/ needs to be also there. But /public/ should be always at root folder. Thank you for the answers!
